nixos/default.nix

```
@@ -65,6 +66,15 @@ in {
         '';
       };
 
+      sharedModules = mkOption {
+        type = types.listOf types.anything;
```

This type seems very general. Perhaps it would be sufficient to restrict the shared modules to file paths? I.e., something like
```suggestion
        type = types.listOf types.path;
```

---

It used to be a `types.path` but I asked for a change because you then cannot use functions.

The verification will be done either way by the module system.

---

If the more general type is necessary then can at least do something like
```suggestion
        type = types.listOf (types.anything // {
          description = "Home Manager module";
        });
```
to make the documentation more clear 🙂

---

For the typing of the extraModules option in a similar PR I wrote(see below), I used something a bit more descriptive: 
`with types; listOf (oneOf [ attrs (functionOf attrs) path ])` so if this one gets merged, I think it should use that.